ktor-server/ktor-server-core/jvm/src/io/ktor/routing/RouteSelector.kt

```diff
@@ -201,15 +217,43 @@ public data class OptionalParameterRouteSelector(val name: String) :
  * Evaluates a route against a constant path segment
  * @param value is a value of the path segment
  */
-public data class PathSegmentConstantRouteSelector(val value: String) :
-    RouteSelector(RouteSelectorEvaluation.qualityConstant) {
-    override fun evaluate(context: RoutingResolveContext, segmentIndex: Int): RouteSelectorEvaluation {
-        if (segmentIndex < context.segments.size && context.segments[segmentIndex] == value)
-            return RouteSelectorEvaluation.ConstantPath
-        return RouteSelectorEvaluation.Failed
+public data class PathSegmentConstantRouteSelector(val value: String) : RouteSelector(RouteSelectorEvaluation.qualityConstant) {
+
+    private var hasTrailingSlash: Boolean = false
+
+    public constructor(value: String, hasTrailingSlash: Boolean): this(value) {
+        this.hasTrailingSlash = hasTrailingSlash
+    }
+
+    override fun evaluate(context: RoutingResolveContext, segmentIndex: Int): RouteSelectorEvaluation = when {
+        segmentIndex == context.segments.lastIndex && hasTrailingSlash != context.hasTrailingSlash ->
+            RouteSelectorEvaluation.Failed
+        segmentIndex < context.segments.size && context.segments[segmentIndex] == value ->
+            RouteSelectorEvaluation.ConstantPath
+        else -> RouteSelectorEvaluation.Failed
     }
 
     override fun toString(): String = value
+
+    // autogenerated
```

Comment:
If we think that the property should be considered, the why don't we simply include it in the primary constructor so we'll be able to omit explicit hashCode/equals implementations?

Reply:
Sorry, this branch was outdated and needed a rebase. This change is obsolete now.